tests/test_version_range.py

```diff
@@ -330,3 +344,18 @@ def test_PypiVersionRange_raises_ivr_for_unsupported_and_invalid_ranges(range, w
             assert expected in str(ivre)
     else:
         assert expected == str(PypiVersionRange.from_native(range))
+
+
+@pytest.mark.parametrize("test_case", open("./tests/data/npm_gitlab.txt").read().splitlines(False))
```

Comment:
If you have a JSON that looks like this:
```
[
    (">=1.0.1 <2.0.1 || >= 3.0.3 < 4.0.4")), "vers:npm/>=1.0.1|<2.0.1|>=3.0.3|<4.0.4"),
    (">=1.0.1 <2.0.1 || >= 3.0.3 < 4.0.4")), "vers:npm/>=1.0.1|<2.0.1|>=3.0.3|<4.0.4"),
    ...
]

```
you could process it with this
```
@pytest.mark.parametrize(["native_gitlab_range", "expected_vers_range", json.load(open("./tests/data/npm_gitlab.json"))
```


And here is a one time thing to create them to run in Python
```
# create the initial files once:
natives = open("./tests/data/npm_gitlab.txt").read().splitlines(False))
import json
from univers.version_range import GitLabVersionRange

test_cases = []
scheme="npm"

for index, native in enumerate(natives, 1):
    vers = GitLabVersionRange.from_gitlab_native(scheme=scheme, string=native)
    test = dict(test_index=index, scheme=scheme, gitlab_native=native, expected_vers=vers)
    test_cases.append(test)

with open(f"tests/data/npm_gitlab.json", "w") as out:
    json.dump(test_cases, out, indent=2)

```
Then your test could become more or less:
```Python
@pytest.mark.parametrize("test_case", json.load(open("./tests/data/pypi_gitlab.json")))
def test_pypi_gitlab_version_range_parse(test_case):
     result = GitLabVersionRange.from_gitlab_native(
        scheme=test_case["scheme"],
        string=test_case["gitlab_native"],
     )
    assert str(result) == test_case["expected_vers"]
```

The benefit is that you are then effectively testing more.
Also the index is to help diagnostic and find which test fails